.github/workflows/appimage.yml

```diff
@@ -0,0 +1,79 @@
+# Manually triggered build of mlnet, mlgui and utils binaries and appimage with mlnet+gui
+
+name: Build AppImage
+
+on:
+  workflow_dispatch
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: sudo apt-get install -y --no-install-recommends camlp4 libnum-ocaml-dev libminiupnpc-dev libnatpmp-dev libbz2-dev librsvg2-dev libmagic-dev libgtk2.0-dev liblablgtk2-ocaml-dev liblablgtk2-gl-ocaml-dev liblablgtk2-gnome-ocaml-dev libgd-dev
+
+      - name: Configure options
+        run: ./configure --enable-batch --enable-upnp-natpmp --disable-directconnect --disable-fasttrack --disable-gnutella --disable-gnutella2 --enable-gui=newgui2
+
+      - name: Build binaries
+        run:  make
+
+      - name: Build utils
+        run:  make utils
+
+      - name: Upload mlnet binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: mlnet
+          path: mlnet
+
+      - name: Upload binaries + utils
+        uses: actions/upload-artifact@v3
+        with:
+          name: Binaries
+          path: |
+                mlnet
+                mlgui
+                bt_dht_node
+                copysources
+                get_range
+                make_torrent
+                mld_hash
+                subconv
+                svg_converter
+          if-no-files-found: warn
+
+      - name: Upload mlnet+gui binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: mlnet+gui
+          path: mlnet+gui
+          if-no-files-found: warn
+
+#  appimage:
+#    needs: build 
+#    runs-on: ubuntu-latest
+#    steps:
+#      - uses: actions/download-artifact@v3
+#        with:
+#          name: mlnet+gui
+
+      - name: Download AppImageBuilder
+        run:  wget https://github.com/AppImageCrafters/appimage-builder/releases/download/v1.0.3/appimage-builder-1.0.3-x86_64.AppImage && chmod +x appimage-builder-1.0.3-x86_64.AppImage
```

Comment:
would be nice to have some way to verify authenticity of this download, because in the end it will build something that we will distribute to users, so need to make sure it comes from the original authors (e.g. check the checksum)

Reply:
A hash of the output file can be computed and added to the generated artifacts without much effort.
Also a digital signature could be added to the workflow if available, but I never tried this option.

Reply:
I mean, there is no other available source for a checksum hash than the embedded sha-1 in the zsync file used by the AppImageUpdate tool, which makes it useless for security purposes since the appimage-builder file comes straight from the horse's mouth in the author's github repo.

Could use something like this:
```
APPIMAGEBUILDER_VERSION=1.1.0
APPIMAGEBUILDER_FILE="appimage-builder-${APPIMAGEBUILDER_VERSION}-x86_64.AppImage"
APPIMAGEBUILDER_URL="https://github.com/AppImageCrafters/appimage-builder/releases/download/v${APPIMAGEBUILDER_VERSION}/${APPIMAGEBUILDER_FILE}"
          wget "${APPIMAGEBUILDER_URL}"
          curl -L "${APPIMAGEBUILDER_URL}.zsync" \
          | sed -ne "1,9s/^sha-1:\s*\([[:xdigit:]]\{40\}\)/\1 *${APPIMAGEBUILDER_FILE}/ip" \
          | sha1sum --warn --check -
```
but don't think that it would be of any help to make sure it comes from the original authors.

Reply:
The hash verification code is added commented since doesn't seems that useful.
Currently the releases of appimage-builder are not signed.
We could add real verification later when and if it becomes available.
https://github.com/AppImageCrafters/appimage-builder/issues/352

Reply:
we download specific version of appimagebuilder, so can hardcode hashsum in the code for that specific version.